src/Cardano/BM/Tracer/Transformers.lhs

```diff
@@ -0,0 +1,46 @@
+\subsection{Cardano.BM.Tracer}
+\label{code:Cardano.BM.Tracer}
+
+%if style == newcode
+\begin{code}
+module Cardano.BM.Tracer.Transformers
+    ( showTracing
+    , condTracing
+    , condTracingM
+    ) where
+
+import           Control.Monad (when)
+import           Data.Functor.Contravariant (Contravariant (..), Op (..))
+
+import           Cardano.BM.Tracer.Class
+
+\end{code}
+%endif
+
+The Tracer transformers exploiting Show.
+
+\begin{code}
+showTracing :: (Show a) => Tracer m String -> Tracer m a
+showTracing = contramap show
+\end{code}
+
+
+The Tracer transformer that allows for on/off control of tracing at
+Trace creation time.
+
+\begin{code}
+condTracing :: (Monad m) => Bool -> Tracer m a -> Tracer m a
+condTracing active tr
+  | active    = tr
+  | otherwise = nullTracer
+\end{code}
+
+The Tracer transformer with dynamic control of tracing.
+
+\begin{code}
+condTracingM :: (Monad m) => m Bool -> Tracer m a -> Tracer m a
```

Comment:
I think a better (and more useful) example would be one with the type signature:

`condTracingM :: (Monad m) => m (a -> Bool) -> Tracer m a -> Tracer m a`

which can be trivially reduced to this case using `const`